File: Problem 1/multiples3and5.clj
;; multiple3and5.clj
;; Written by James Wang
;; Project Euler Problem 1

;; Find the sum of all the multiples of 3 or 5 below 1000

(defn ans [x y acc]
  (if (>= x y)
    acc
    (if (or (= (mod x 3) 0) (= (mod x 5) 0))
      (ans (+ x 1) y (+ acc x))
      (ans (+ x 1) y acc))))

(println (ans 3 1000 0))

File: Problem 1/multiples3and5.py
# multiples3and5.py
# Written by James Wang
# Project Euler Problem 1

# Find the sum of all the multiples of 3 or 5 below 1000.

ans = sum(n for n in range(1000) if n % 3 == 0 or n % 5 == 0)
print(ans)
